components/script/dom/navigator.rs

```diff
@@ -171,11 +190,16 @@ impl NavigatorMethods for Navigator {
 
     // https://www.w3.org/TR/gamepad/#navigator-interface-extension
     fn GetGamepads(&self) -> DomRoot<GamepadList> {
-        let root = self
-            .gamepads
-            .or_init(|| GamepadList::new(&self.global(), &[]));
+        let global = self.global();
```

Comment:
I would expect some wpt tests for these changes

Reply:
The existing WPT that covers this fails on an assert that getGamepads() returns an array. Currently it returns a GamepadList, and attempting to change it to a frozen array of gamepads is non-trivial due to `to_frozen_array` seemingly being incompatible with the Gamepad object.

Reply:
This is now covered by the aforementioned WPT test (`gamepad/not_fully_active.https.html`)